### PR TITLE
[WOR-871] Fix Sam CI workflow dispatch paths

### PR DESCRIPTION
**Pull request by aherbst-broad**
Ticket: <Link to Jira ticket>

  \<Don't forget to include the ticket number in the PR title!\>

What:

  \<For your reviewers' sake, please describe in a sentence or two what this PR is accomplishing (usually from the users' perspective, but not necessarily).\>

Why:

  \<For your reviewers' sake, please describe in ~1 paragraph what the value of this PR is to our users or to ourselves.\>

How:

  \<For your reviewers' sake, please describe in ~1 paragraph how this PR accomplishes its goal.\>

  \<If the PR is big, please indicate where a reviewer should start reading it (i.e. which file or function).\>

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

**Review by gpolumbo-broad — APPROVED** [review]

**Comment by gpolumbo-broad** [comment]
This looks like the right thing to do to fix it.  Can you convert this to a "real" PR?

**Comment by aherbst-broad** [comment]
@gpolumbo-broad will do as soon as this is passing checks.

File: .github/workflows/sam-build-tag-publish-and-run-tests.yaml [hunk 1]
```diff
@@ -103,7 +103,7 @@ jobs:
       - name: dispatch to terra-github-workflows
         uses: aurelien-baudet/workflow-dispatch@93e95b157d791ae7f42aef8f8a0d3d723eba1c31 #commit sha for v2.1.1
```
Comment (ichengchang) [inline .github/workflows/sam-build-tag-publish-and-run-tests.yaml]: Replace `aurelien-baudet/workflow-dispatch@93e95b157d791ae7f42aef8f8a0d3d723eba1c31 #commit sha for v2.1.1` with the newer `broadinstitute/workflow-dispatch@v3`. The newer version is less buggy and avoid using deprecated GitHub library functions.

File: .github/workflows/sam-build-tag-publish-and-run-tests.yaml [hunk 2]
```diff
@@ -125,7 +125,7 @@ jobs:
...
```
Comment (ichengchang) [inline .github/workflows/sam-build-tag-publish-and-run-tests.yaml]: Replace `aurelien-baudet/workflow-dispatch@93e95b157d791ae7f42aef8f8a0d3d723eba1c31 #commit sha for v2.1.1` with the newer `broadinstitute/workflow-dispatch@v3`. The newer version is less buggy and avoid using deprecated GitHub library functions.

**Review by ichengchang — CHANGES_REQUESTED** [review]
Left some comments.

**Comment by aherbst-broad** [comment]
jenkins retest

**Review by ichengchang — APPROVED** [review]
This looks good!

**Pull request merged**